src/main/scala/nodes/images/Convolver.scala

```diff
@@ -101,29 +101,31 @@ object Convolver {
       whitener: Option[ZCAWhitener],
       varConstant: Double): DenseMatrix[Double] = {
     var x,y,chan,pox,poy,py,px = 0
-    while (x < resWidth) {
-      y = 0
-      while (y < resHeight) {
-        chan = 0
-        while (chan < imgChannels) {
-          poy = 0
-          while (poy < convSize) {
-            pox = 0
-            while (pox < convSize) {
+
+    poy = 0
+    while (poy < convSize) {
```

Comment:
Doing this is super annoying and error-prone. I'm normally allergic to macros, but this seems like the kind of thing we could make fast with meta-programming. Any ideas?